docker/createml.go

```diff
@@ -106,9 +107,13 @@ func PutManifestList(a *types.AuthInfo, yamlInput types.YAMLInput, ignoreMissing
 			img.Platform.OSFeatures = imgMfst.OSFeatures
 		}
 
+		if imgMfst.Variant != "" {
```

Comment:
There is no variant detail in a Docker v2 or OCI image config, so this field will never be filled in (see the `makeImageInspect()` function that fills this in from a fetch of the manifest/config). So, I don't think adding `Variant` to the struct makes sense or this particular "if not empty" block. The rest of this is fine.

Reply:
I'll have to check the spec, but it's definitely how different arm variations are detected and is necessary, just inspect any library image. 

After testing it looks like this can be omitted, the variant gets added to the inputyaml and populated properly. Sending update shortly.

Reply:
Oh, image manifest, nvm. Commit inbound.